mscore/workspace.cpp

```diff
@@ -76,26 +76,41 @@ static QString editedWorkspaceTranslatableName(const QString& oldWorkspaceTransl
 
 static QString defaultWorkspaceTranslatableName(const QString& editedWorkspaceName)
       {
-      const auto it = std::find(defaultEditedWorkpaces.begin(), defaultEditedWorkpaces.end(), editedWorkspaceName);
+      const auto it = std::find(WorkspacesManager::defaultEditedWorkspaces.begin(), WorkspacesManager::defaultEditedWorkspaces.end(), editedWorkspaceName);
 
-      if (it != defaultEditedWorkpaces.end()) {
-            const int idx = it - defaultEditedWorkpaces.begin();
-            if (idx < int(defaultWorkspaces.size()))
-                  return defaultWorkspaces[idx];
+      if (it != WorkspacesManager::defaultEditedWorkspaces.end()) {
+            const int idx = it - WorkspacesManager::defaultEditedWorkspaces.begin();
+            if (idx < int(WorkspacesManager::defaultWorkspaces.size()))
+                  return WorkspacesManager::defaultWorkspaces[idx];
             }
 
       return QString();
       }
 
-//---------------------------------------------------------
-//   undoWorkspace
-//---------------------------------------------------------
-
-void MuseScore::undoWorkspace()
+/**
+ *   Reverts current workspace to the state of the `source` workspace
+ */
+void MuseScore::resetWorkspace()
       {
-      // TODO: make a separate session start backup?
-      WorkspacesManager::currentWorkspace()->read();
-      WorkspacesManager::currentWorkspace()->setDirty(false);
+      //if currentWorkspace is the `edited` Basic or Advanced one, remove the edited and show the source one
+      if (WorkspacesManager::isDefaultEditedWorkspace(WorkspacesManager::currentWorkspace())) {
+            //Basic edited
+            const QString& currWorkspaceName = WorkspacesManager::currentWorkspace()->translatableName();
+            if (currWorkspaceName == WorkspacesManager::defaultEditedWorkspaces[0]) {
+                  WorkspacesManager::remove(WorkspacesManager::currentWorkspace());
+                  changeWorkspace(WorkspacesManager::findByTranslatableName(WorkspacesManager::defaultWorkspaces[0]));
```

Comment:
The correct name can be found with [`defaultWorkspaceTranslatableName`](https://github.com/musescore/MuseScore/blob/9d1f5955b8fc76101eba8d0bb7f414e6735a7222/mscore/workspace.cpp#L77) function, then these conditions checks could be avoided.